src/ucm/util/replace.h

```diff
@@ -27,6 +27,23 @@ extern pthread_t volatile ucm_reloc_get_orig_thread;
 #define UCM_DEFINE_REPLACE_FUNC(_name, _rettype, _fail_val, ...) \
     _UCM_DEFINE_REPLACE_FUNC(ucm_override_##_name, ucm_##_name, _rettype, _fail_val, __VA_ARGS__)
 
+#if defined(__riscv)
+#define _UCM_DEFINE_REPLACE_FUNC(_over_name, _ucm_name, _rettype, _fail_val, ...) \
+    \
+    /* Define a symbol which goes to the replacement - in case we are loaded first */ \
+    _rettype _over_name(UCM_FUNC_DEFINE_ARGS(__VA_ARGS__)) \
```

Comment:
it seems like only the difference here is ucm_trace call ? What is the problem with the trace call ?

Reply:
Sorry I missed the above discussion